### PR TITLE
Rate limiter tweaks

### PR DESCRIPTION
**Pull request by ob6160**
## What does this change?

* Truncates the access token that we log in the rate limiter middleware when the limit is hit. This is so that we don't exceed the maximum message size in Kibana
* Checks that the global bucket key array isn't empty.
* Extends middleware test to check that the correct rate limit exceeded messages are being logged.
* Removes some unnecessary newlines

### CODE REVIEW

File: src/server/lib/middleware/rateLimit.ts [hunk 1]
```diff
@@ -65,9 +65,14 @@ export const rateLimiterMiddleware = async (
   });
 
   if (ratelimitBucketTypeIfHit) {
+    // Cut down the size of the access token so we don't make the log line too long.
+    const truncatedAccessToken = rateLimitData.accessToken
+      ? rateLimitData.accessToken.substring(0, 6)
+      : '';
```
Comment (liamness) [inline src/server/lib/middleware/rateLimit.ts:71]: Could do `rateLimitData.accessToken?.substring(0, 6)` here maybe

File: src/server/routes/verifyEmail.ts [hunk 2]
```diff
@@ -146,7 +144,6 @@ router.post(
 
 router.get(
   '/verify-email/:token',
-
```
Comment (liamness) [inline src/server/routes/verifyEmail.ts:149]: Do we know what was up with this formatting before? Some kind of mass find + replace operation maybe?

I see that `yarn test` doesn't run prettier. Was that why this wasn't caught?

**Review by liamness — APPROVED** [review]
Looks good to me, only minor comments.

File: src/server/routes/verifyEmail.ts (hunk 2 above)
Comment (ob6160) [reply src/server/routes/verifyEmail.ts:149]: I think it was from when I moved from having the `rateLimitMiddleware` included at this level to moving it up to the router so we didn't have to add it to every single route. Odd that the linter didn' catch it

File: src/server/lib/middleware/rateLimit.ts (hunk 1 above)
Comment (ob6160) [reply src/server/lib/middleware/rateLimit.ts:71]: good idea!

File: src/server/routes/verifyEmail.ts (hunk 2 above)
Comment (liamness) [reply src/server/routes/verifyEmail.ts:149]: I think only ESLint runs on the CI, not Prettier.

File: src/server/lib/middleware/rateLimit.ts [hunk 3]
```diff
@@ -66,9 +66,10 @@ export const rateLimiterMiddleware = async (
 
   if (ratelimitBucketTypeIfHit) {
     // Cut down the size of the access token so we don't make the log line too long.
-    const truncatedAccessToken = rateLimitData.accessToken
-      ? rateLimitData.accessToken.substring(0, 6)
-      : '';
+    const truncatedAccessToken =
+      typeof rateLimitData.accessToken === 'string'
+        ? rateLimitData.accessToken?.substring(0, 6)
```
Comment (liamness) [inline src/server/lib/middleware/rateLimit.ts:71]: There won't be any need for the optional chaining syntax here as the type is known.

File: src/server/routes/verifyEmail.ts (hunk 2 above)
Comment (ob6160) [reply src/server/routes/verifyEmail.ts:149]: 👍

File: src/server/lib/middleware/rateLimit.ts (hunk 3 above)
Comment (ob6160) [reply src/server/lib/middleware/rateLimit.ts:71]: 👍

**Pull request merged**